packages/psss-server/src/connections/ReportConnection.ts

```diff
@@ -9,4 +9,15 @@ const { REPORT_API_URL = 'http://localhost:8030/v2' } = process.env;
 
 export class ReportConnection extends ApiConnection {
   baseUrl = REPORT_API_URL;
+
+  async fetchReport(reportCode: string, orgUnitCodes: string[], periods: string[] = []) {
+    if (!orgUnitCodes || !orgUnitCodes.length) {
+      throw new Error(`No organisationUnitCodes provided`);
+    }
+
+    return this.post(`fetchReport/${reportCode}`, {
```

Comment:
I definitely find it weird that the fetching of data uses a `post`!

Reply:
I think it was because initially we authenticate the requests using username/password (stored in the request body). So it was changed from a get to a post. Now we no longer do that but if we send test data (which added by @rohan-bes ) then we also need to store it in the request body

Reply:
Ok maybe that test data thing should be a special case - I think in general this should just be a `GET` with an Authorization header

Reply:
@edmofro @chris-bes sums it up correctly! Now I believe we can switch back to `fetchReport` being a `GET` endpoint, and we can switch the `testData`/`testConfig` behaviour on to a separate endpoint entirely (which would still be `POST`) 👍